plugins/content/document/lib/operation_engines/KOperationEngineImageMagick.php

```diff
@@ -124,17 +126,21 @@ protected function operationComplete($rc, $output) {
 		}
 	}
 	
-	// The returned xml will be stored in the images directory. it than can be downloaded by he user with serveFlavorAction and provide him
+	// The returned xml will be stored in the images directory. it than can be downloaded by the user with serveFlavorAction and provide him
 	// information about the created images.
-	private function createImagesListXML($imagesList){
+	private function createImagesListXML($imagesList, $outDirPath, $key){
 		sort($imagesList);
+		$i = 1;
 		$imagesListXML = new SimpleXMLElement('<'.self::IMAGES_LIST_XML_LABEL_ITEMS.'/>');
 		foreach ($imagesList as $image) {
     		$imageNode = $imagesListXML->addChild(self::IMAGES_LIST_XML_LABEL_ITEM);
-    		$imageNode->addChild(self::IMAGES_LIST_XML_LABEL_NAME, $image);
+		$imageNode->addAttribute(self::IMAGES_LIST_XML_ATTRIBUTE_INDEX, $i++);
```

Comment:
why do we need this attribute?

Reply:
To make sure we have alignment between the metadata file and the imageList file. As before, we could compare the file by name - but if the name encrypted - we should have indexes to align them